core/src/main/scala/org/dbpedia/extraction/dataparser/DateTimeParser.scala

```diff
@@ -74,29 +75,201 @@ class DateTimeParser ( context : {
 
     private val YearRegex = ("""(?iu)""" + prefix + """(?<![\d\pL\w])(-?\d{1,4})(?!\d)\s*(""" + eraRegex + """)?""" + postfix).r
 
+    // get Date object from the node converted into string
+    def getHeidelTime(text: String, datatype: Datatype): Option[Date] = {
+        val heidelTime = new HeidelTimeStandalone(
+            de.unihd.dbs.uima.annotator.heideltime.resources.Language.ENGLISH,
+            DocumentType.NARRATIVES,
+            OutputType.TIMEML,
+            "src/main/resources/config.props",
+            POSTagger.TREETAGGER, true)
 
-    override def parse(node : Node) : Option[Date] =
+        val result = heidelTime.process(text)
+        return convTimeMLtoDate(result, datatype)
+
+
+    }
+
+    // Check if the array contains only numbers apart from the string XXXX
+    def check(date_array : Array[String]) : Boolean = {
+        for ( x <- date_array){
+            if( x != "XXXX"){
+                if( !x.forall(_.isDigit) ){
+                    return false
+                }
+            }
+        }
+        return true
+    }
+
+    // Converts the output of Heidel Parser which is in TimeX3 format into Date
+    // for eg 2007-04-21 , XXXX-4-2
+    def convTimeMLtoDate(text: String, datatype: Datatype) : Option[Date] =
```

Comment:
Oh yeah I can probably use it to parse TimeX3 format too. I shall have a look into it.